### PR TITLE
Adding unit tests

### PR DESCRIPTION
**Pull request by ziccardi**
## Motivation

JIRA: https://issues.jboss.org/browse/AGDROID-752

## Description

Adding unit tests

## Progress

- [x] Change OIDCAuthenticatorImpl to use DI
- [x] Fix wrong packages
- [x] Add unit tests for auth module
- [x] Add unit tests for code module

### CODE REVIEW

File: auth/src/main/java/org/aerogear/mobile/auth/authenticator/AuthenticationServiceFactory.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            return authorizationService;
+        }
+
+        public AuthState getAuthState() {
+            return authState;
+        }
+    }
+
+    public AuthenticationServiceFactory(final Context appContext) {
+        this.appContext = appContext;
```
Comment (secondsun) [inline auth/src/main/java/org/aerogear/mobile/auth/authenticator/AuthenticationServiceFactory.java]: Do a safety getApplicationContext().

File: auth/src/main/java/org/aerogear/mobile/auth/authenticator/AuthenticationServiceFactory.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public ServiceWrapper createAuthorizationService(final KeycloakConfiguration keycloakConfiguration,
```
Comment (secondsun) [inline auth/src/main/java/org/aerogear/mobile/auth/authenticator/AuthenticationServiceFactory.java]: javadocs

File: auth/src/main/java/org/aerogear/mobile/auth/authenticator/AuthenticationServiceFactory.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import org.aerogear.mobile.auth.configuration.KeycloakConfiguration;
+
+public class AuthenticationServiceFactory {
+
+    private final static String SCOPE_OPENID = "openid";
+
+    private final Context appContext;
+
+
+    public static class ServiceWrapper {
```
Comment (secondsun) [inline auth/src/main/java/org/aerogear/mobile/auth/authenticator/AuthenticationServiceFactory.java]: javadocs

File: auth/src/main/java/org/aerogear/mobile/auth/authenticator/AuthenticationServiceFactory.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import net.openid.appauth.AuthorizationServiceConfiguration;
+import net.openid.appauth.ResponseTypeValues;
+import net.openid.appauth.browser.BrowserBlacklist;
+import net.openid.appauth.browser.VersionedBrowserMatcher;
+
+import org.aerogear.mobile.auth.ConnectionBuilderForTesting;
+import org.aerogear.mobile.auth.configuration.AuthServiceConfiguration;
+import org.aerogear.mobile.auth.configuration.KeycloakConfiguration;
+
+public class AuthenticationServiceFactory {
```
Comment (secondsun) [inline auth/src/main/java/org/aerogear/mobile/auth/authenticator/AuthenticationServiceFactory.java]: javadocs

File: auth/src/main/java/org/aerogear/mobile/auth/authenticator/OIDCAuthenticatorImpl.java [hunk 5]
```diff
@@ -52,23 +43,28 @@
     private final KeycloakConfiguration keycloakConfiguration;
     private final AuthServiceConfiguration authServiceConfiguration;
 
-    private final Context appContext;
+    //private final Context appContext;
```
Comment (secondsun) [inline auth/src/main/java/org/aerogear/mobile/auth/authenticator/OIDCAuthenticatorImpl.java]: Why comment when you can remove?

Comment (ziccardi) [reply auth/src/main/java/org/aerogear/mobile/auth/authenticator/OIDCAuthenticatorImpl.java]: It is a WIP, but thanks ;)

**Comment by matzew** [comment]
LGTM

**Pull request merged**